src/arkode/arkode_mristep_nls.c

```diff
@@ -449,7 +457,7 @@ int mriStep_NlsLSolve(N_Vector b, void* arkode_mem)
      Fsi(z) (stored step_mem->Fsi[step_mem->istage])
      r = zc - gamma*Fsi(z) - step_mem->sdata
   ---------------------------------------------------------------*/
-int mriStep_NlsResidual(N_Vector zcor, N_Vector r, void* arkode_mem)
+SUNNlsStatus mriStep_NlsResidual(N_Vector zcor, N_Vector r, void* arkode_mem)
```

Comment:
Again, the values returned by this function should be converted to valid `SUNNlsStatus` constants